app/src/components/organisms/SurveyForm.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import Stack from '@mui/material/Stack';
+import SurveyClassQuestion from 'components/organisms/SurveyClassQuestion';
+import Button from '@mui/material/Button';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import TextField from '@mui/material/TextField';
+import Divider from '@mui/material/Divider';
+import { classData, formVals, vals } from '../shared/interfaces/surveyForm.interfaces';
+import { overallDefaults } from '../shared/constants/surveyForm.constants';
+
+function SurveyForm(props: { formData: classData[] }) {
+  const [disable, setDisabled] = useState(true);
+  const [values, setValues] = useState(() => {
+    const currentValues: formVals = {
+      relief: false,
+      explanation: '',
+      courses: {}
+    };
+
+    currentValues.courses = props.formData.reduce((obj: any, field) => {
+      obj[field.CourseID.subject + ' ' + field.CourseID.code] = {
+        ...overallDefaults
+      };
+      return obj;
+    }, {});
+
+    return currentValues;
+  });
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    console.log(values);
+    e.preventDefault();
+    //TODO: submit values somewhere
+  };
+
+  const fieldChanged = (courseName: string, type: string, value: string) => {
+    setValues((currentValues) => {
+      currentValues.courses[courseName][type as keyof vals] = value;
+      return currentValues;
+    });
+  };
+
+  const handleCheckbox = (event: React.ChangeEvent<HTMLInputElement>, checkbox: boolean) => {
+    setValues((currentValues) => {
+      if (checkbox) {
+        currentValues.relief = event.target.checked;
+        setDisabled(!disable);
+      } else {
+        currentValues.explanation = event.target.value;
+      }
+      return currentValues;
+    });
+  };
+
+  return (
+    <div>
+      <form onSubmit={onSubmit}>
+        <Stack style={{ height: '500px', overflowY: 'scroll' }} spacing={4}>
+          {Object.keys(values.courses).map((field) => {
+            return <SurveyClassQuestion key={field} name={field} fieldChanged={fieldChanged} />;
+          })}
+        </Stack>
+        <Divider />
+        <Stack spacing={4}>
+          <FormControlLabel
+            control={<Checkbox />}
+            label="Have Relief?"

```

Comment:
I don't understand what this question means, probably needs more detail. I see that you also disable the corresponding text entry unless this is checked which is great, but I would also hide the box `display:none` unless the box is checked.

Reply:
We could also wrap this field in a <Tooltip> to provide extra information without cluttering the page.

Reply:
Had a conversation with Ben about this. Course Relief is essentially a pass to take one less course per relief. Changed to accepting a number rather than a checkbox to reflect this.